From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/aeb8696ffa4b
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: MSI: also reject resource with flags all clear

When resource assignment fails, things may end up this way, and we want
to avoid using the resource in that case.

Derived from an upstream patch by Yijing Wang <wangyijing@huawei.com>.

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- a/drivers/pci/msi-xen.c
+++ b/drivers/pci/msi-xen.c
@@ -204,7 +204,9 @@ static u64 find_table_base(struct pci_de
 	bar = reg & PCI_MSIX_FLAGS_BIRMASK;
 
 	flags = pci_resource_flags(dev, bar);
-	if (flags & (IORESOURCE_DISABLED | IORESOURCE_UNSET | IORESOURCE_BUSY))
+	if (!flags ||
+	     (flags & (IORESOURCE_DISABLED | IORESOURCE_UNSET |
+		       IORESOURCE_BUSY)))
 		return 0;
 
 	return pci_resource_start(dev, bar);
